common/src/main/java/io/pravega/common/util/NewtonianSearch.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright Pravega Authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.common.util;
+import java.util.function.Function;
+
+/**
+ * TO-DO.
+ */
+public class NewtonianSearch {
+    @SuppressWarnings("unchecked")
```

Comment:
These warnings that are being suppressed are real errors in the code. Please don't ignore lints like this.